xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -420,7 +386,7 @@ public void onReportResponse(long reportIntervalNano) {
 
         // TODO(zdapeng): Use XdsClient to do Lrs directly.
         // For now create an LRS Client.
-        if (xdsConfig.balancerName != null) {
+        if (channel != null) {
```

Comment:
`channel` will never be null here.

Reply:
`channel` is null for non-EDS only case. The `channel` field is only for EDS only.

Reply:
No, since you have not migrated to use `XdsClientImp` (`XdsClient` in this class is really `XdsComms2`), you are still creating channel for non-EDS only case.

Btw, "non-EDS only case" sounds very weird...

Reply:
If the `XdsClientRef` is available in the attributes, `LookasideLb` will not create a channel, instead, it will used the `XdsClient` from the Ref directly, regardless of its `XdsComms2` or `XdsClientImpl`

Reply:
Oh, I missed that. You are right.